lib/impress.log.js

```diff
@@ -234,20 +229,14 @@ impress.log.mixin = (application) => {
         application.log.write(fileType, message);
         impress.log.write(fileType, '[' + application.name + ']\t' + message);
       };
-      let fn;
-      if (application.isImpress) {
-        if (!serverLog) fn = fnN;
-        else fn = fnI;
-      } else if (!serverLog && !applicationLog) {
-        fn = fnN;
-      } else if (!serverLog && applicationLog) {
-        fn = fnA;
-      } else if (serverLog && !applicationLog) {
-        fn = fnI;
-      } else if (serverLog && applicationLog) {
-        fn = fnAI;
-      }
-      application.log[fileType] = fn;
+      application.log[fileType] = (
```

Comment:
It became much harder to read now (it was not easy before as well), you might as well rewrite it like that:
```javascript
const logFunctions = [fnN, fnA, fnI, fnAI, fnN, fnN, fnI, fnI];
let index = application.isImpress << 2 |
            serverLog << 1 |
            applicationLog << 0;
application.log[fileType] = logFunctions[index];
```
It does not add clarity, just feels like a more neat way to write conditions looking like this one.
I don't really know if there is any better way to do this, so that it is clear what this condition does at all, because I still don't understand it.